src/main/java/edu/umich/tl/CourseDelete.java

```diff
@@ -25,57 +31,101 @@
+		 *  
+		 */
+		for (String canvasTermId : previousTerms) {
+			CoursesForDelete coursesForDelete=new CoursesForDelete();
+			getUnpublishedCourseList(canvasTermId,coursesForDelete,apiHandler,null);
+			//Todo delete below line
+			M_log.debug("CourseList: "+coursesForDelete.getCourses().size()+" term: "+canvasTermId);
+		}
 	}
 
 	public enum CanvasCallEnum{
 		 API_DIRECT_CANVAS, 
 		 API_ESB_CANVAS
 	}
-
+     /*
+      * exit code '1' implies something I expected could potentially go wrong
+      * http://stackoverflow.com/questions/2434592/difference-in-system-exit0-system-exit-1-system-exit1-in-java 
+      */
 	private static void readPropertiesFromFile(String[] args) {
 		M_log.debug("readPropertyFiles(): called");
 		Properties properties = null;
 		String propFileLocation;
 		if(args.length == 0) {
 			M_log.error("Command line arguments are not provided");
-			System.exit(0);
+			System.exit(1);
 		}
 		propFileLocation = args[0];
 		properties = Utils.getPropertiesFromURL(propFileLocation);
 		canvasToken = properties.getProperty(CANVAS_TOKEN);
 		canvasURL = properties.getProperty(CANVAS_URL);
 		esbURL=properties.getProperty(ESB_URL);
-		sisTermIdFromPropsFile = properties.getProperty(SIS_TERM_ID);
+		getTermsInfoFromProperties(properties);
 		checkForApiType(properties);
 	}
+	
+	/*
+	 * The Terms that needs for course deletion are listed in the properties file. Each term has 2 piece of information 
+	 * term1=2030;04/06/15, the sisTermCode and the endDate after which the courses should be considered for deletion.
+	 * We will check if the current system date is past the endDate then that term becomes eligible for deleting the 
+	 * unused/unpublished courses that are created as apart of auto-provisioning of courses in canvas
+	 */
+	private static void getTermsInfoFromProperties(Properties properties) {
+		M_log.debug("getTermsInfoFromProperties(): called");
+		String termCount = properties.getProperty(TERM_COUNT);
+		if(termCount.isEmpty()) {
+			M_log.error("The property \"term.count\" is not set in the properties file \"canvasCourseDelete.properties\"");
+			System.exit(1);
+		}
+		int termCountInt = Integer.parseInt(termCount);
+		for(int i=1;i<=termCountInt;i++) {
+			String termPropertyName = TERM+String.valueOf(i);
+			String termPropertyValue = properties.getProperty(termPropertyName);
+			if(Utils.isEmpty(termPropertyValue)) {
+				M_log.error("The property \""+termPropertyName+"\" don't seems to exist in \"canvasCourseDelete.properties\" "
+						+ "please make sure all term properties are numerically aligned and number of properties is equal to \"term.count\" property value");
+				System.exit(1);
+			}
+			String[] termWithDate = termPropertyValue.split(";");
+			DateTimeFormatter dtf = DateTimeFormat.forPattern("MM/dd/yyyy");
+			String sisTermId = termWithDate[0];
+			DateTime endDate = dtf.parseDateTime(termWithDate[1]);
```

Comment:
Avoid creating variables or keeping object instances around if they're only used one time, like `dtf`.  There are several ways to clean this up.  Instead of this:

```java
DateTimeFormatter dtf = DateTimeFormat.forPattern("MM/dd/yyyy");
String sisTermId = termWithDate[0];
DateTime endDate = dtf.parseDateTime(termWithDate[1]);
```

Try one of these:

1. Shorten it to one line, without declaring `dtf`:

  ```java
DateTime endDate = DateTimeFormat.forPattern("MM/dd/yyyy").parseDateTime(termWithDate[1]);
```

1. Avoid redeclaring variables in a loop.  It *might* waste time constructing and disposing the objects repeatedly.  Declare them before the loop and use them within:

  ```java
DateTime endDate;
DateTimeFormatter dateFormatter = DateTimeFormat.forPattern("MM/dd/yyyy");
for(int i=1; i<=termCountInt; i++) {
// ...
endDate = dateFormatter.parseDateTime(termWithDate[1]);
// ...
}
```

  In addition to those two variables in this example, move all other variable declarations out of loops, too.  (E.g. `termPropertyName`, `termPropertyValue`, `termWithDate`, etc.) If their value doesn't change within the loop, initialize them, too.

*PS:  It turns out that "dtf" is a rude bit of Internet slang and isn't a very good choice for a variable name.  Everybody appreciates descriptive names anyway.*